packages/zsa-react/src/results.ts

```diff
@@ -95,9 +95,9 @@ export const calculateResultFromState = <
     return {
       isPending: true,
       isOptimistic: false,
-      data: undefined,
+      data: result.data,
       isError: false,
-      error: undefined,
+      error: result.error,
       isSuccess: false,
       status: "pending",
     }
```

Comment:
I wonder if this `if` and the next one can be merged into one. I don't have all the context here on `oldResult.status` `empty` or `filled`. 
I'm sure I'm missing something but the UI looks so much better with this change as you can see in the videos : )